File: 2021/golang/src/day1/main.go
package main

import (
	"strconv"
	"utils"
)

var data = getData(utils.GetLines("input.txt"))

func getData(lines []string) []int {
	var retVal []int
	for _, e := range lines {
		i, _ := strconv.Atoi(e)
		retVal = append(retVal, i)
	}
	return retVal
}

func partOne() string {
	count := 0
	for i := 1; i < len(data); i++ {
		if data[i] > data[i-1] {
			count += 1
		}
	}

	return strconv.Itoa(count)
}

func partTwo() string {
	count := 0
	for i := 3; i < len(data); i++ {
		if data[i]+data[i-1]+data[i-2] > data[i-1]+data[i-2]+data[i-3] {
			count += 1
		}
	}

	return strconv.Itoa(count)
}

func main() {
	utils.RunWithTimer(partOne) // 1766 -- took 0 ms
	utils.RunWithTimer(partTwo) // 1797 -- took 0 ms
}

File: 2021/python/day1/main.py
from aoc_utils import run_with_timer

data = [int(x.strip()) for x in open("input.txt").readlines()]


def part_one():
	return sum(1 for i in range(1, len(data)) if data[i] > data[i-1])


def part_two():
	return sum(1 for i in range(3, len(data)) if data[i] + data[i-1] + data[i-2] > data[i-1] + data[i-2] + data[i-3])


if __name__ == '__main__':
	run_with_timer(part_one)  # 1766 -- took 0 ms
	run_with_timer(part_two)  # 1797 -- took 0 ms
